framework/Optimizers/SimulatedAnnealing.py

```diff
@@ -0,0 +1,718 @@
+      conv.addSub(InputData.parameterInputFactory(name, contentType=InputTypes.FloatType,descr=descr,printPriority=108  ))
+
+    # Presistance
+    conv.addSub(InputData.parameterInputFactory('persistence', contentType=InputTypes.IntegerType,
+        printPriority = 109,
+        descr=r"""provides the number of consecutive times convergence should be reached before a trajectory
+              is considered fully converged. This helps in preventing early false convergence."""))
+
+    # Cooling Schedule
+    coolingSchedule = InputData.parameterInputFactory('coolingSchedule',
+        printPriority=109,
+        descr=r""" The function governing the cooling process. Currently, user can select between,"""
+                  # \xmlString{linear},
+                  +r"""\xmlString{exponential},
+                  \xmlString{cauchy},
+                  \xmlString{boltzmann},"""
+                  # \xmlString{fast},
+                  +r"""or \xmlString{veryfast}.\\ \\"""
+                  #In case of \xmlString{linear} is provided, The cooling process will be governed by: $$ T^{k} = T^0 - 0.1 * k$$
+                  +r"""In case of \xmlString{exponential} is provided, The cooling process will be governed by: $$ T^{k} = T^0 * \alpha^k$$
+                  In case of \xmlString{boltzmann} is provided, The cooling process will be governed by: $$ T^{k} = \frac{T^0}{log(k + d)}$$
+                  In case of \xmlString{cauchy} is provided, The cooling process will be governed by: $$ T^{k} = \frac{T^0}{k + d}$$"""
+                  #In case of \xmlString{fast} is provided, The cooling process will be governed by: $$ T^{k} = T^0 * \exp(-ck)$$
+                  +r"""In case of \xmlString{veryfast} is provided, The cooling process will be governed by: $$ T^{k} =  T^0 * \exp(-ck^{1/D}),$$
+                  where $D$ is the dimentionality of the problem (i.e., number of optimized variables), $k$ is the number of the current iteration
+                  $T^{0} = \max{(0.01,1-\frac{k}{\xmlNode{limit}})}$ is the initial temperature, and $T^{k}$ is the current temperature
+                  according to the specified cooling schedule.
+                  \default{exponential}.""")
+    specs.addSub(coolingSchedule)
+
+    for schedule,param in cls.coolingOptions.items(): # FIXME: right now this allows multiple cooling schedule, which should be fixed as soon as
+                                                              # InputData can allow having list of subnodes
+      sch = InputData.parameterInputFactory(schedule, contentType = InputTypes.StringType,descr=schedule+' cooling schedule')
+      for par,descr in param.items():
+        sch.addSub(InputData.parameterInputFactory(par, contentType = InputTypes.FloatType,descr=descr))
+      coolingSchedule.addSub(sch)
+    return specs
+
+  @classmethod
+  def getSolutionExportVariableNames(cls):
+    """
+      Compiles a list of acceptable SolutionExport variable options.
+      @ In, None
```

Comment:
we don't document `self` though, either, because when you call this you don't provide that argument; I think `cls` and `self` are the same principle in this case?

`names = Optimizers.SimulatedAnnealing.getSolutionExportVariableNames()` is the valid call I think.